onnxruntime/test/providers/cpu/math/einsum_test.cc

```diff
@@ -595,9 +595,11 @@ TEST(Einsum, ImplicitEinsumAsTensorContraction) {
 // Theme: Half support
 
 TEST(Einsum, ExplicitEinsumAsIdentity_1D_input_Half) {
+#if !defined(USE_WEBGPU)
```

Comment:
Is it better if we move `#if !defined(USE_WEBGPU)` check into `HasCudaEnvironment`? It's noisy to scatter so many places.

Reply:
I also think it is not good that there are many `#if !defined(USE_WEBGPU)`. But I think it is also unreasonable move it into a CUDA special file. Is there Any other good ideas from anyone?